### PR TITLE
feat: enhance backend test setup

### PR DESCRIPTION
**Pull request by emcclosky**
# Yurt

## Changes
- Refactor server implementation to achieve parity between the actual server and test server
- Add Http service and tests
- Implement tests for user query and mutation
- Add jest config and refactor use of `import-meta-url` for compatibility 

## Notes
This update should hopefully give a nice foundation to more complex test scenarios in our backend. Will and Patrick have already taken a look, but I wanted to make sure everyone had an opportunity to raise any questions or offer thoughts.

### CODE REVIEW

File: packages/backend/docker-compose.jest.yml
Comment (will-t-harris) [inline packages/backend/docker-compose.jest.yml:1]: We have the normal database's `docker-compose.yml` at the root of the monorepo, would it make sense for this to live there as well?

File: packages/backend/jest.config.mjs [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  },
+  transform: {
+    '^.+\\.tsx?$': [
+      'ts-jest',
+      {
+        tsconfig: '<rootDir>/tsconfig.json',
+        isolatedModules: true,
+        useESM: true,
+        diagnostics: {
+          ignoreCodes: [1343],
```
Comment (will-t-harris) [inline packages/backend/jest.config.mjs:24]: Do we still need to ignore this code with the `ts-jest-mock-import-meta` library added?

**Review by will-t-harris — APPROVED** [review]
All looks good to me, great work Liz! 👏 

Just a couple questions below.

**Review by pfuscoe — APPROVED** [review]

File: packages/backend/jest.config.mjs (hunk 1 above)
Comment (emcclosky) [reply packages/backend/jest.config.mjs:24]: Yep, it's mentioned in the `ts-jest-mock-import-meta` docs that in order for the transformer to work we need to ignore error code 1343.

File: packages/backend/docker-compose.jest.yml
Comment (emcclosky) [reply packages/backend/docker-compose.jest.yml:1]: Location updated!

**Review by jonnyschult — APPROVED** [review]

**Pull request merged**